

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Frequently Asked Questions &mdash; FALCON 0.5 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="author" title="About these documents"
              href="about.html"/>
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="FALCON 0.5 documentation" href="index.html"/>
        <link rel="next" title="Tertiary" href="tertiary.html"/>
        <link rel="prev" title="&lt;no title&gt;" href="glossary.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> FALCON
          

          
          </a>

          
            
            
              <div class="version">
                0.5
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About Falcon</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_start.html">Quick Start Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="pipeline.html">Pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands.html">Commands</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="parameters.html">Parameters</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Frequently Asked Questions</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#general">General</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#can-i-start-from-corrected-reads">Can I start from corrected reads?</a></li>
<li class="toctree-l3"><a class="reference internal" href="#how-do-i-select-a-length-cutoff">How do I select a length cutoff?</a></li>
<li class="toctree-l3"><a class="reference internal" href="#what-s-the-difference-between-a-primary-and-an-associated-contig">What&#8217;s the difference between a Primary and an Associated contig?</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#what-are-the-differences-between-a-ctg-fasta-and-a-ctg-base-fasta">What are the differences between a_ctg.fasta and a_ctg_base.fasta</a></li>
<li class="toctree-l4"><a class="reference internal" href="#for-a-given-contig-in-a-ctg-fa-how-can-i-find-it-s-primary-contig-map-coordinates">For a given contig in a_ctg.fa, how can I find it&#8217;s primary contig map coordinates?</a></li>
<li class="toctree-l4"><a class="reference internal" href="#how-does-falcon-avoid-chimeras-given-homologous-repeat-regions-an-different-chromosomes">How does FALCON avoid chimeras given homologous repeat regions an different chromosomes?</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#can-falcon-handle-x-ploid-genome-data">Can Falcon handle X-ploid genome data?</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#workflow">Workflow</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#how-do-i-restart-a-failed-workflow">How do I restart a failed workflow?</a></li>
<li class="toctree-l3"><a class="reference internal" href="#how-do-i-turn-on-logging">How do I turn on logging?</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tertiary.html">Tertiary</a></li>
<li class="toctree-l1"><a class="reference internal" href="references.html">References</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">FALCON</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Frequently Asked Questions</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/faq.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="frequently-asked-questions">
<span id="faq"></span><h1>Frequently Asked Questions<a class="headerlink" href="#frequently-asked-questions" title="Permalink to this headline">¶</a></h1>
<div class="section" id="general">
<h2>General<a class="headerlink" href="#general" title="Permalink to this headline">¶</a></h2>
<div class="section" id="can-i-start-from-corrected-reads">
<h3>Can I start from corrected reads?<a class="headerlink" href="#can-i-start-from-corrected-reads" title="Permalink to this headline">¶</a></h3>
<p>Yes. The option <a class="reference internal" href="parameters.html#input-type"><span class="std std-ref">input_type</span></a> can be set to either <code class="docutils literal"><span class="pre">raw</span></code> or <code class="docutils literal"><span class="pre">preads</span></code>. In the case of the latter,
<a class="reference internal" href="cmds/fc_run.html#fc-run-py"><span class="std std-ref">fc_run.py</span></a> will assume the fasta files in <a class="reference internal" href="parameters.html#input-fofn"><span class="std std-ref">input_fofn</span></a> are all error-corrected reads and it
will ignore any <a class="reference internal" href="glossary.html#term-error-correction"><span class="xref std std-term">error correction</span></a> step and go directly into the final assembly overlapping step.</p>
</div>
<div class="section" id="how-do-i-select-a-length-cutoff">
<h3>How do I select a length cutoff?<a class="headerlink" href="#how-do-i-select-a-length-cutoff" title="Permalink to this headline">¶</a></h3>
<p>The option <a class="reference internal" href="parameters.html#length-cutoff"><span class="std std-ref">length_cutoff</span></a> controls the read length cutoff used during the
<a class="reference internal" href="glossary.html#term-error-correction"><span class="xref std std-term">error correction</span></a> process and <a class="reference internal" href="parameters.html#length-cutoff-pr"><span class="std std-ref">length_cutoff_pr</span></a> controls the cutoff used for the
final assembly overlapping steps. In the final assembly, more reads may not lead to a better assembly due to the
fact that some of the reads can be noisy and create false links in the assembly graph. Sometimes you might want
to re-run the final steps of the assembly pipeline in <code class="docutils literal"><span class="pre">2-asm-falcon</span></code> with different values for <code class="docutils literal"><span class="pre">--min_len</span></code>
in <code class="docutils literal"><span class="pre">run_falcon_asm.sub.sh</span></code> as this step is quick relative to the overlap detection steps in the earlier stages
of the pipeline.</p>
<p>If you&#8217;re not sure, and you are not compute resource limited, one strategy is to choose a smaller
<a class="reference internal" href="parameters.html#length-cutoff"><span class="std std-ref">length_cutoff</span></a> and do the computation once. Later, one can use a different
<a class="reference internal" href="parameters.html#length-cutoff-pr"><span class="std std-ref">length_cutoff_pr</span></a> to achieve a more contiguous assembly</p>
<p>In general we recommend that you tune the cutoff so that you&#8217;re left with roughly 15x to 20x for final genome assembly.
If you set <a class="reference internal" href="parameters.html#length-cutoff"><span class="std std-ref">length_cutoff</span></a> equal to <code class="docutils literal"><span class="pre">-1</span></code>, FALCON will attempt to autocalculate this cutoff
for you.</p>
</div>
<div class="section" id="what-s-the-difference-between-a-primary-and-an-associated-contig">
<span id="primary-vs-associated"></span><h3>What&#8217;s the difference between a Primary and an Associated contig?<a class="headerlink" href="#what-s-the-difference-between-a-primary-and-an-associated-contig" title="Permalink to this headline">¶</a></h3>
<p><a class="reference internal" href="glossary.html#term-primary-contig"><span class="xref std std-term">Primary contigs</span></a> can be thought of as the longest continuous stretches of contiguously
assembled sequence, while <a class="reference internal" href="glossary.html#term-associated-contig"><span class="xref std std-term">associate contigs</span></a> can be thought of mostly as structural
variants that occur over the length of the primary contigs. Thus, each alternate primary contig configuration
(associated contig) can be &#8220;associated&#8221; with it&#8217;s primary based on it&#8217;s <code class="docutils literal"><span class="pre">XXXXXXF</span></code> prefix.</p>
<p>Some basic information about how the associated contigs are generated can be found
in <a class="reference external" href="https://speakerdeck.com/jchin/string-graph-assembly-for-diploid-genomes-with-long-reads">this speakerdeck</a> (pg.14)
, <a class="reference external" href="https://speakerdeck.com/jchin/learning-genome-structrues-from-de-novo-assembly-and-long-read-mapping">here</a>
(pg.14-15) <a class="reference external" href="https://speakerdeck.com/jchin/learning-genome-structrues-from-de-novo-assembly-and-long-read-mapping">and here</a>.</p>
<p>Conceptually, if a genome is haploid, then all contigs should be primary contigs. However, in general there will usually
still be some associated contigs generated. This is likely due to:</p>
<ol class="arabic simple">
<li>Sequencing errors</li>
<li>Segmental duplications.</li>
</ol>
<p>For the first case, Quiver should help by filtering out low quality contigs. Since there is more sequence in
the set of primary contigs for blasr to anchor reads and there is no true unique region in the erroneous
associated contigs, the raw read coverage on them should be low. We can thus filter low quality
<a class="reference internal" href="glossary.html#term-associated-contig"><span class="xref std std-term">associated contig</span></a> consensus as there won&#8217;t be much raw read data to support them.</p>
<p>For the second case, one could potentially partition the reads into different haplotype groups and construct
an assembly graph for each haplotype and generate contigs accordingly.</p>
<p>If a genome is a diploid genome, then most of the associated contigs will be locally alternative alleles.
Typically, when there are big structural variations between homologous chromosomes, there will be alternative
paths in the assembly graph and the alternative paths correspond to the associated contigs. In such case,
the primary contigs are “fused contigs” from both haplotypes.</p>
<p>FALCON_unzip is currently being developed to resolve the haplotypes so <a class="reference internal" href="glossary.html#term-haplotig"><span class="xref std std-term">haplotigs</span></a> can
be generated. Two videos illustrating the concept - (<a class="reference external" href="https://youtu.be/yC1ujdLUT7Q">Video 1</a> ,
<a class="reference external" href="https://youtu.be/vwSyD31eahI">Video 2</a>)</p>
<p>A <a class="reference external" href="https://twitter.com/infoecho/status/604070162656985088">slide</a> illustrating the method on a synthetic genome.</p>
<div class="section" id="what-are-the-differences-between-a-ctg-fasta-and-a-ctg-base-fasta">
<h4>What are the differences between a_ctg.fasta and a_ctg_base.fasta<a class="headerlink" href="#what-are-the-differences-between-a-ctg-fasta-and-a-ctg-base-fasta" title="Permalink to this headline">¶</a></h4>
<p>The file <code class="docutils literal"><span class="pre">a_ctg_base.fasta</span></code> contains the sequences in the primary contigs fasta that correspond to the associated
contigs inside <code class="docutils literal"><span class="pre">a_ctg.fasta</span></code>. Namely, each sequence of a_ctg_base.fasta is a contiguous sub-sequence of a primary
contig. For each sequence inside <a href="#id1"><span class="problematic" id="id2">`</span></a>a_ctg_base.fasta, there are one or more associated contigs in a_ctg.fasta.</p>
</div>
<div class="section" id="for-a-given-contig-in-a-ctg-fa-how-can-i-find-it-s-primary-contig-map-coordinates">
<h4>For a given contig in a_ctg.fa, how can I find it&#8217;s primary contig map coordinates?<a class="headerlink" href="#for-a-given-contig-in-a-ctg-fa-how-can-i-find-it-s-primary-contig-map-coordinates" title="Permalink to this headline">¶</a></h4>
<p>The 2nd field and the 3rd field of the sequence header inside <code class="docutils literal"><span class="pre">a_ctg.fa</span></code> indicate the begin node and the end node of
the contig. For example, if we have a header like</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>&gt;000000F-001-01 000941458:E 000486369:E <span class="m">15593</span> <span class="m">47559</span> <span class="m">5</span> 0.9969 0.8447
</pre></div>
</div>
<p>It means the associated contig 000000F-001-01 starts from node 000941458:E and ends at 000486369:E. Thsee two nodes
should be also in the path of the corresponding primary contig. The path of the primary contig is fully specified in
the file p_ctg_tiling_path, you can find exact beginning and ending points where the associated contig are attached
to the primary contigs. However, the coordinates are not conserved after the <span class="xref std std-ref">Quiver</span> consensus step,
it might be necessary to do some quite alignment to recalibrate the attaching points after quiver consensus.
In some case, you can even just do quick sequence alignment to find the homologous region in the primary contig of
an associated contigs.</p>
</div>
<div class="section" id="how-does-falcon-avoid-chimeras-given-homologous-repeat-regions-an-different-chromosomes">
<h4>How does FALCON avoid chimeras given homologous repeat regions an different chromosomes?<a class="headerlink" href="#how-does-falcon-avoid-chimeras-given-homologous-repeat-regions-an-different-chromosomes" title="Permalink to this headline">¶</a></h4>
<p>Such repeats are typically called as “segmental duplications”. Yes, Falcon will collapse these regions if the
overlapper can not distinguish the repeats. As discussed above in some case, it is just
like the case of a diploid genome, we can potentially resolve the two distinct haplotypes. In other cases,
the repeat is more complicated, such as if there are more than 2 copies, (e.g. the middle part of contigs 4006 in
page 21 of
<a class="reference external" href="https://speakerdeck.com/jchin/learning-genome-structrues-from-de-novo-assembly-and-long-read-mapping">this slide deck</a>.
To resolve these regions, we&#8217;ll need to do more investigation to separate the reads into more than two groups
to resolve them.</p>
</div>
</div>
<div class="section" id="can-falcon-handle-x-ploid-genome-data">
<h3>Can Falcon handle X-ploid genome data?<a class="headerlink" href="#can-falcon-handle-x-ploid-genome-data" title="Permalink to this headline">¶</a></h3>
<p>Falcon, in its current form, is a “diploid or polyploid aware assembler”. I believe there is no fully specific
definition what a “diploid or polyploid assembler” should deliver yet at the moment of this writing.
From the point of the genome assembly research field, it is still quite new. There were a couple of paper published
before for diploid assemblies. However, the general strategy is the phasing adding reads on top on earlier assembly
step.</p>
<p>To some degree, the current Falcon assembler provides a better way to build that foundation for a full diploid /
polyploid assembler. Please refer to this slide deck
<a class="reference external" href="https://speakerdeck.com/jchin/string-graph-assembly-for-diploid-genomes-with-long-reads">https://speakerdeck.com/jchin/string-graph-assembly-for-diploid-genomes-with-long-reads</a> for some detail. Some
technical details of the deck are already obsoleted for a little bit, but the general concept is still applied to
most recent code in Falcon.</p>
<p>For a tetraploid genome, depending on the genome structure, I would argue one will get better continuity from
the primary contigs if you use Falcon for assembling the genome. However, you will need to do good analysis
on both primary and associated contigs (or better, the assembly graph directly) after running Falcon to
interpret the results correctly. The primary contigs will be “fused” contigs from all haplotypes unless
the differences between haplotypes are big such that the assembler’s overlap segregate them apart already.</p>
<p>There are some prototype work to fully segregate the “fused primary contigs” for diploid case. I just
presented the ideas in #SFAF2015 conference. For tetraploid case, it will need some hard-code non-trivial
mathematics research work to get it work right.</p>
</div>
</div>
<div class="section" id="workflow">
<h2>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline">¶</a></h2>
<div class="section" id="how-do-i-restart-a-failed-workflow">
<h3>How do I restart a failed workflow?<a class="headerlink" href="#how-do-i-restart-a-failed-workflow" title="Permalink to this headline">¶</a></h3>
<p>Often times restarting a FALCON job due to unexplained failure is an easy process. It&#8217;s typically just a matter
of removing any sentinel files and re-invoking <code class="docutils literal"><span class="pre">fc_run.py</span> <span class="pre">fc_run.cfg</span></code> from the FALCON root directory. Read
<a class="reference external" href="https://github.com/PacificBiosciences/FALCON/wiki/Tips">this</a> section of the wiki for details.
If your job failed due to
<a class="reference external" href="https://github.com/PacificBiosciences/FALCON/wiki/Contrib#disk-quotas">quota or other disk full issues</a>
, you may need to wipe the directory and start over again due to corrupt DB&#8217;s</p>
</div>
<div class="section" id="how-do-i-turn-on-logging">
<h3>How do I turn on logging?<a class="headerlink" href="#how-do-i-turn-on-logging" title="Permalink to this headline">¶</a></h3>
<p>See <a class="reference external" href="https://github.com/PacificBiosciences/FALCON/issues/139">this</a> github issue.</p>
<p>In short, you should pass <code class="docutils literal"><span class="pre">logging.ini</span></code> as the 2nd argument to <a class="reference internal" href="commands.html#fc-run-py"><span class="std std-ref">fc_run.py</span></a></p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>$ fc_run.py fc_run.cfg logging.ini
</pre></div>
</div>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tertiary.html" class="btn btn-neutral float-right" title="Tertiary" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="glossary.html" class="btn btn-neutral" title="&lt;no title&gt;" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016, Greg Concepcion, Sarah Kingan, Chris Dunn, Jason Chin.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.5',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>